orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -296,3 +296,11 @@ def failed_processes() -> List[str]:
             logger.info("Subscription already in sync")
     except ValueError as e:
         raise_status(HTTPStatus.NOT_FOUND, str(e))
+
+
+@router.get("/{subscription_id}/metadata", response_model=None, status_code=HTTPStatus.OK)
+def subscription_metadata(subscription_id: UUID, response: Response) -> Optional[dict]:
+    md = get_subscription_metadata(subscription_id)
+    if md is None:
+        response.status_code = HTTPStatus.NOT_FOUND
```

Comment:
Of course the question is always if you want to return nothing (an empty dict) or a 404.  I'm slightly in favour of an empty dictionary since optionals always introduce additional checking.

Reply:
What I think makes most sense is:
* If the subscription exists -> return 200 with body `metadata or {}`
* If the subscription does not exist -> return 404

(I recall similar discussions on the customer descriptions endpoint, last year or so :) )